usb_flash_drive_copy.py

```diff
@@ -0,0 +1,543 @@
+            f"sudo umount {self.mount_point}", shell=True
+        )
+        if return_code != 0:
+            raise UsbFlashDriveUnmountableError(
+                f"Unable to unmount USB flash drive from '{self.mount_point}'!"
+            )
+        log.write("USB flash drive unmounted.")
+
+
+class OTCameraUsbCopier(Observer):
+    def __init__(
+        self,
+        power_led: Led,
+        wifi_led: Led,
+        rec_led: Led,
+        src_dir: Path,
+        usb_flash_drive: UsbFlashDrive,
+    ) -> None:
+        """This classes' main purpose is to provide methods to copy and delete videos
+        with the help of a `CopyInformation` object.
+
+        Args:
+            power_led (Led): give user visual feedback to inform about the current
+            status of the OTCameraUsbCopier.
+            wifi_led (Led): give user visual feedback to inform about the current
+            status of the OTCameraUsbCopier.
+            rec_led (Led): give user visual feedback to inform about the current
+            status of the OTCameraUsbCopier.
+            src_dir (Path): directory where the video files to be copied are located at.
+            usb_flash_drive (UsbFlashdrive): The USB flash drive.
+        """
+        self.power_led = power_led
+        self.wifi_led = wifi_led
+        self.rec_led = rec_led
+        self.src_dir = src_dir
+        self.usb_flash_drive = usb_flash_drive
+
+    def update(self, subject: Subject) -> None:
+        if isinstance(subject, Button):
+            if not subject.is_active:
+                log.write("Shutdown OTCamera.")
+                self.shutdown()
+
+    def shutdown(self):
+        """Shutdown OTCamera."""
+
+        self._turn_off_all_leds()
+        self.rec_led.blink(n=4, background=False)
+
+        self._turn_off_all_leds()
+
+        if not config.DEBUG_MODE_ON:
+            log.closefile()
+            subprocess.call("sudo shutdown -h now", shell=True)
+
+    def _turn_off_all_leds(self):
+        """Turn off all LEDs."""
+        self.power_led.turn_off()
+        self.wifi_led.turn_off()
+        self.rec_led.turn_off()
+
+    def copy_to_usb(self, copy_info: CopyInformation) -> None:
+        """Copy over videos to USB flash drive.
+
+        The WiFi LED blinking indicates the videos being copied over.
+        The WiFi LED constantly being on indicates that the copy process is finished.
+
+        Args:
+            copy_info (CopyInformation): the copy information.
+        """
+        self.wifi_led.blink()
+        log.write("Start copying files")
+        for video in copy_info.videos:
+            if video.copied:
+                log.write(f"Video at: '{ video.path}' already copied. Skipping.")
+                continue
+
+            if not video.path.exists():
+                log.write(
+                    f"Video at: '{ video.path}' does not exists.", log.LogLevel.WARNING
+                )
+                continue
+
+            shutil.copy2(
```

Comment:
Could there be an exception that interrupts video processing?
In such a case, the log messages are not written and the `wifi_led.turn_on()` will not be called. The user will not be notified that the copy process finished?